plans/caching.go

```diff
@@ -0,0 +1,116 @@
+package plans
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"reflect"
+	"strings"
+
+	"github.com/memcachier/mc"
+)
+
+const defaultCacheExpirationTime = 0 // never expire
+
+func prepareForCache(data interface{}) (string, error) {
+	var b bytes.Buffer
+	w := gzip.NewWriter(&b)
+	err := json.NewEncoder(w).Encode(convertToMemcacheVal(data))
+	if err != nil {
+		w.Close()
+		return "", err
+	}
+	err = w.Flush()
+	if err != nil {
+		w.Close()
+		return "", err
+	}
+	err = w.Close()
+	if err != nil {
+		return "", err
+	}
+	return string(b.Bytes()), nil
+}
+
+func restoreFromCache(encoded string, targetType reflect.Value) ([]interface{}, error) {
+	r, err := gzip.NewReader(strings.NewReader(encoded))
+	if err != nil {
+		r.Close()
+		return nil, err
+	}
+
+	typed := reflect.Zero(reflect.SliceOf(targetType.Type())).Interface()
+	err = json.NewDecoder(r).Decode(&typed)
+	if err != nil {
+		return nil, err
+	}
+	err = r.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	tv := reflect.ValueOf(typed)
+	data := reflect.MakeSlice(reflect.SliceOf(reflect.TypeOf([]interface{}{})), tv.Len(), tv.Cap()).Interface().([]interface{})
```

Comment:
I think this should be `reflect.MakeSlice(reflect.TypeOf([]interface{}{}), tv.Len(), tv.Cap()).Interface()` - the `reflect.SliceOf` call seems unnecessary.